### PR TITLE
Av 1059/magnotometer

### PR DESCRIPTION
**Pull request by adeshkumar1**
Added magnetometer functionality based off of TARS

### CODE REVIEW

File: MIDAS/src/sensor_data.h [hunk 1]
```diff
@@ -78,10 +78,10 @@ struct GPS {
     float satellite_count = 0;
 };
 
-struct Magnetometer {
-    float mx = 0;
-    float my = 0;
-    float mz = 0;
+struct MagnetometerReading {
```
Comment (redindelible) [inline MIDAS/src/sensor_data.h:81]: Why was this renamed?

File: MIDAS/src/sensor_data.h [hunk 2]
```diff
@@ -78,10 +78,10 @@ struct GPS {
...
+    float mx;
```
Comment (redindelible) [inline MIDAS/src/sensor_data.h:82]: Why were the default values removed?

File: MIDAS/src/sensors.h [hunk 3]
```diff
@@ -38,11 +38,19 @@ struct OrientationSensor {
     Orientation read();
 };
 
+struct Magnetometer {
```
Comment (redindelible) [inline MIDAS/src/sensors.h:41]: Why was this renamed?

File: MIDAS/src/rocket_state.h [hunk 4]
```diff
@@ -42,7 +42,7 @@ struct RocketState {
     SensorState<Continuity> continuity;
     SensorState<Voltage> voltage;
     SensorState<GPS> gps;
-    SensorState<Magnetometer> magnometer;
+    SensorState<MagnetometerReading> magnometer;
```
Comment (redindelible) [inline MIDAS/src/rocket_state.h:45]: We should probably fix the spelling of `magnometer` at some point.

**Review by redindelible — CHANGES_REQUESTED** [review]
Nothing is particularly broken, but there are some places the code can be improved.

File: MIDAS/src/sensor_data.h (hunk 1 above)
Comment (NPIPHI) [reply MIDAS/src/sensor_data.h:81]: Name conflict with the sensor

File: MIDAS/src/sensor_data.h (hunk 2 above)
Comment (NPIPHI) [reply MIDAS/src/sensor_data.h:82]: Default values disallow struct initialization

File: MIDAS/src/sensors.h (hunk 3 above)
Comment (NPIPHI) [reply MIDAS/src/sensors.h:41]: It didn't exist before

**Review by aidancostello — APPROVED** [review]
